instrument/utils/utilities.py

```diff
@@ -0,0 +1,446 @@
+
+
+def show_constraints():
+    """
+    Show constraints and freeze angles (value)
+    """
+    _geom_ = current_diffractometer()
+    _geom_.show_constraints()
+
+
+def reset_constraints():
+    """
+    Reset all constraints
+    """
+    _geom_ = current_diffractometer()
+    _geom_.reset_constraints()
+    _geom_.show_constraints()
+
+
+def set_constraints():
+    """
+    Change constraint values for specific axis
+    """
+    _geom_ = current_diffractometer()
+    axes = _geom_.calc._engine.engine.axis_names_get(0)
+    for axis in axes:
+        low = _geom_.get_axis_constraints(axis).low_limit
+        high = _geom_.get_axis_constraints(axis).high_limit
+        angle = _geom_.get_axis_constraints(axis).value
+        value = (
+            input(
+                "{} constraints low, high = [{:3.3f}, {:3.3f}]: ".format(
+                    axis, low, high
+                )
+            )
+        ) or [low, high]
+        if isinstance(value, str):
+            value = value.replace(",", " ").split(" ")
+        _geom_.apply_constraints(
+            {axis: Constraint(value[0], value[1], angle, True)}
+        )
+    _geom_.show_constraints()
+
+
+def change_diffractometer(*args):
+    _geom_ = current_diffractometer()
+    list = ["diffract", "fourc", "psic"]
+    print("Available diffractometers {}".format(list))
+    diff = input("Diffractometer ({}): ".format(_geom_.name)) or _geom_.name
+    if diff == "diffract":
+        select_diffractometer(diffract)
+    elif diff == "fourc":
+        select_diffractometer(fourc)
+    elif diff == "psic":
+        select_diffractometer(psic)
+    else:
+        raise ValueError("Diffractometer type {} not existing.".format(diff))
+
+    _geom_ = current_diffractometer()
+    print("Diffractometer changed to {}".format(_geom_.name))
+
+
+def plotselect(detector=None):
+    """
+    Selects scalers plotted during scan
+    """
+
+    scalers = scaler.channels_name_map.items()
+    plotted_scalers = scaler.hints["fields"]
+    print("{:>4}{:>12}{:>4}".format("#", "Detector", " "))
+    det_list = []
+    for num, item in enumerate(scalers):
+        if item[0] in plotted_scalers:
+            print("{:>4}{:>12}{:>4}".format(num, item[0], "<--"))
+            det_list.append(num)
+        else:
+            print("{:>4}{:>12}".format(num, item[0]))
+    dets = input("Scalers to be plotted {}: ".format(det_list)) or det_list
+    if isinstance(dets, str):
+        dets = [int(x) for x in dets.split(",")]
+    det_list = []
+    for num, item in enumerate(scalers):
+        if num in dets:
+            det_list.append(item[0])
+    print("Scalers plotted: {}".format(det_list))
+    scaler.select_plot_channels(det_list)
+
+
+def set_counting_time(time=None, monitor=False):
+    """
+    Sets counting time / monitor counts:
+        time <  200: counting time in seconds
+        time >= 200: monitor counts
+
+    Needs to be adapted to other detectors than scalers
```

Comment:
Every detector will have a `.preset_monitor` that will change the counting time in that detector, for example for the lambda: https://github.com/APS-4ID-POLAR/polar_instrument/blob/566c5d7dc6a7c85abd9d9baf095da0e2b6728ab1/instrument/devices/4idd/ad_lambda.py#L133